packages/compiler/core/util.ts

```diff
@@ -492,3 +493,101 @@ export class DuplicateTracker<K, V> {
     }
   }
 }
+
+export function createRekeyableMap<K, V>(entries?: [K, V][]): RekeyableMap<K, V> {
```

Comment:
I guess I am never sure on when and why we do `create` function vs `class` but here wouldn't it make more sense to align with `Map`?

Reply:
I had trouble exporting class from this file with interface in types. Map is both an interface and a type and I would like to keep types.ts more self-contained and mostly interfaces. I assume there is a way to satisfy the compiler. If you know what I'm missing, happy to change this.

Reply:
ah I see, yeah I guess that's maybe better this way, you can cheat by having both the same name but then if its in different file it can be trouble